python/IECore/ParameterParser.py

```diff
@@ -488,6 +496,15 @@ def __serialiseStringArray( parameter, value ) :
 
 	return list(value)
 
+def __serialiseStringArrayWithEscape( parameter, value ) :
+
+	# when serialising string arrays that can start with "-"
+	# we take care to escape it with a leading "\",
+	# so it won't be confused with a new argument name
+	# we also escape anything that starts with any number of "\" followed by a "-"
+	# so they don't confused with an intentional escape
+	return [ "\\{}".format( x ) if re.match( r"^\\*-.*", x ) else x for x in value ]
```

Comment:
Won't this break the serialisation of arguments for `acceptsFlags = True` parameters? On L360-362 above, you can see that we just parse those verbatim, so now we will be getting `\-` where we would have been getting `-` before.

Reply:
Good catch. Addressed here: 48e06c95c71f5fd22e757b5f5eeda5a09a1d8f40

But I think we only need it now for backward compatibility, right? If you go through the serialisation (and therefore the escaping) process, there should be no need to ever have the `acceptsFlags = True` anymore. Or am I missing something?

Reply:
Thanks Ivan. We use `acceptFlags = True` for several of the apps in Gaffer, and we don't want users to have to start escaping them manually on the command line. That wouldn't have been affected by this PR because the serialiser isn't involved, but it is a reason to continue using `acceptsFlags`. Longer term I want to drop `IECore.Parameter` from app arguments and use a `Plug/argparse` combo, but that's some way off.

Reply:
Right. That makes sense. Thanks!